codetable/src/hasher_impl.rs

```diff
@@ -21,7 +17,6 @@ macro_rules! derive_write {
     };
 }
 
-#[cfg(any(feature = "blake2b", feature = "blake2s"))]
```

Comment:
I think we are somewhat conflating two things here:

1. A set of hashes for building a custom codetable
2. A ready-to-use custom codetable

The need for the (1) could be mitigated if we would tackle https://github.com/multiformats/rust-multihash/issues/179 because it would allow people to use the individual hash crates directly instead of having to implement our `Hasher` trait.

Have you tried whether a blanket impl over `T: Digest` works?

Reply:
> Have you tried whether a blanket impl over `T: Digest` works?

Argh that doesn't work because the two `finalize` functions are not compatible.

Reply:
I think retaining the feature-flags is a good middleground for now.

Reply:
> I think retaining the feature-flags is a good middleground for now.

Done.